ckanext/datastore/tests/test_search.py

```diff
@@ -928,10 +929,9 @@ def test_self_join(self):
             where a.author = b.author
             limit 2
             '''.format(self.data['resource_id'])
-        data = {'sql': query}
-        postparams = json.dumps(data)
+        data = urllib.urlencode({'sql': query})
```

Comment:
why is this change required?

Reply:
I can't remember. Probably a change in how Flask encodes special characters in query args. I'll have a look.

Reply:
Flask is a bit more picky than Pylons when parsing POST data (with good reason).

In these tests we are doing calls via the webtest app like this:

```python
        res = self.app.post('/api/action/datastore_search_sql', params=postparams,
                              extra_environ=auth
        )
```

Which by default uses `Content-type: 'multipart/form-data'`. But what we are actually sending is JSON (see removed lines).

So we either send Form data (with `urllib.urlencode()` like here) or we use the appropriate content type header:

```python
        res = self.app.post('/api/action/datastore_search_sql', params=postparams,
                              extra_environ=auth,
                              headers={'Content-type': 'application/json'})
```

Reply:
And this particular test failed because of the equal sign in the params

Reply:
thanks that makes sense! I was just concerned that we broke one of the ways of calling the api.